### PR TITLE
8250799: NumberStringConverter and its subclasses are missing documentation for all their constructors

### PR DESCRIPTION
**Pull request by nlisker**
This was missed the in Javadoc fixes for 15.

Added documentation for all the constructors, did a bit of formatting, and removed redundant comments.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8250799](https://bugs.openjdk.java.net/browse/JDK-8250799): NumberStringConverter and its subclasses are missing documentation for all their constructors

### Reviewers
 * Johan Vos ([jvos](@johanvos) - **Reviewer**)
 * Kevin Rushforth ([kcr](@kevinrushforth) - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/277/head:pull/277`
`$ git checkout pull/277`

### CODE REVIEW

File: modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java [hunk 1]
```diff
@@ -32,8 +32,7 @@
 import javafx.util.StringConverter;
 
 /**
- * <p>{@link StringConverter} implementation for {@link Number} values
- * that represent currency.</p>
+ * A {@link StringConverter} implementation for {@link Number} values that represent currency. Instances of this class are immutable.
```
Comment (kevinrushforth) [inline modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java:35]: Given that there are no setters or other mutator methods, the immutability seems obvious. I don't see a problem with stating it explicitly, though. It's up to you.

File: modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java [hunk 2]
```diff
@@ -42,30 +41,45 @@
...
-    // ------------------------------------------------------------ Constructors
+    /**
+     * Constructs a {@code CurrencyStringConverter} with the default locale and format.
+     */
     public CurrencyStringConverter() {
         this(Locale.getDefault());
     }
 
+    /**
+     * Constructs a {@code CurrencyStringConverter} with the given locale and the default format.
```
Comment (kevinrushforth) [inline modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java:52]: Can you also add the `@param locale` tag? This applies to all of the newly documented constructors that have arguments.

File: modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java (hunk 1 above)
Comment (nlisker) [reply modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java:35]: True, but I just looked at [Border](https://openjfx.io/javadoc/14/javafx.graphics/javafx/scene/layout/Border.html) which states immutability, so I did the same. I don't mind changing it.

Comment (kevinrushforth) [reply modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java:35]: I don't mind either way.

Comment (johanvos) [reply modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java:35]: I don't think it's bad to stating this explicitly here.

Comment (kevinrushforth) [reply modules/javafx.base/src/main/java/javafx/util/converter/CurrencyStringConverter.java:35]: Agreed.

**Pull request closed without merging**